assets/scripts/fe/src/app/components/projects/project-details/project-details.component.ts

```diff
@@ -0,0 +1,111 @@
+import { Component, OnInit, HostListener } from '@angular/core';
+import { StateService } from '@uirouter/angular';
+import { StandupService } from '../../../commons/services/history/standup.service'
+import { NavService } from '../../../commons/services/utils/nav.service';
+
+@Component({
+  selector: 'app-project-details',
+  templateUrl: './project-details.component.html',
+  styleUrls: ['./project-details.component.css']
+})
+export class ProjectDetailsComponent implements OnInit {
+
+  constructor(
+    private standupservice : StandupService,
+    private state          : StateService,
+    private nav            : NavService,
+  ) 
+  {
+    // nav configuration
+    // TODO: this sucks!. find a better solution
+    this.nav.setNav('Project Details', true);
+  }
+
+  ngOnInit() {
+
+    if (!this.standupservice.noreload && this.state.$current.name !== 'project-details-report') {
```

Comment:
refactor. if and elif contains the same items except `setDateRange()`. there should be a cleaner way to this

Reply:
search negation in programming